lib/Resque/Worker.php

```diff
@@ -531,8 +533,9 @@ public function workerPids()
      */
     public function registerWorker()
     {
-        Resque::redis()->sadd('workers', (string)$this);
-        Resque::redis()->set('worker:' . (string)$this . ':started', strftime('%a %b %d %H:%M:%S %Z %Y'));
+        $this->workerPing();
+        Resque::redis()->sadd(Resque::WORKERS, (string)$this);
```

Comment:
shouldnt we make it atomic? like a transaction or something to prevent broken states.